middleware.go

```diff
@@ -0,0 +1,107 @@
+package jwt
+
+import (
+	"errors"
+	"github.com/Digital-Insight-Technologies-Ltd/gin-jwt-cognito/models"
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+)
+
+// UserClaimsKey is the key for user claims in our context
+const UserClaimsKey = "userClaims"
+
+// CognitoClaimsMiddleware is a middleware for Gin that extracts user claims from
+// AWS Cognito.
+func CognitoClaimsMiddleware(jwtParser *jwt.Parser) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		authHeader := c.GetHeader("Authorization")
+		if !strings.HasPrefix(authHeader, "Bearer ") {
```

Comment:
Make the "Bearer " prefix check case-insensitive to accommodate different client implementations. Use `strings.HasPrefix(strings.ToLower(authHeader), "bearer ")`. [medium]